apps/src/craft/agent/craft.js

```diff
@@ -712,7 +712,13 @@ export default class Craft {
     });
 
     // Run user generated code, calling appCodeOrgAPI
-    const code = Blockly.getWorkspaceCode();
+    let code = '';
+    let codeBlocks = Blockly.mainBlockSpace.getTopBlocks(true);
```

Comment:
would it be worthwhile to move this code into a wrapper function so we don't have to repeat it? Does the reference to `studioApp().initializationBlocks` mean we can't move this into the wrapper?

Reply:
Potentially... although the changes in these craft files are really just a straight revert of https://github.com/code-dot-org/code-dot-org/commit/3c8902bd93dd86fc183976d7abf178ed19987f91

There's quite a few variations of the logic here in the various lab files, so I think it would take some thought to figure out exactly what we want to abstract and move to the wrapper.

Reply:
👍 makes sense